wlm/src/main/java/ai/djl/serving/wlm/SageMakerUtils.java

```diff
@@ -0,0 +1,130 @@
+        }
+        catch (Exception exception) {
+            logger.error("Parsing metadata.yaml failed");
+            throw new ModelException("Parsing metadata yaml failed");
+        }
+
+        if (!validateMetaData(metaDataMap)) {
+            // If metadata is not a valid format, try native DJL serving infer engine.
+            return null;
+        }
+
+        // To validate both schema and inferenceSpec
+        boolean customizedSchema = hasCustomizedSchema(metaDataMap);
+        boolean customizedInferenceSpec = hasCustomizedInferenceSpec(metaDataMap);
+        if (customizedSchema || customizedInferenceSpec) {
+            // For either customized schema or customized inference spec using python engine with sagemaker entry point
+            logger.info("Customized schema builder or inference spec is detected, using python backend");
+            prop.setProperty("option.entryPoint", "djl_python.sagemaker");
+            return "Python";
+        }
+
+        if (metaDataMap.get(MODEL_TYPE).equals(XGBOOST_MODEL_TYPE)) {
+            return "XGBoost";
+        } else if (metaDataMap.get(MODEL_TYPE).equals(PYTORCH_MODEL_TYPE)) {
+            return "PyTorch";
+        } else {
+            logger.error("Invalid model type: " + metaDataMap.get(MODEL_TYPE));
+            throw new ModelException(String.format(
+                    "Model type %s is not supported in SageMaker model format yet", metaDataMap.get(MODEL_TYPE).toString()));
+        }
+    }
+
+    private static Boolean validateMetaData(Map<String, Object> metaDataMap) {
+        // Required field for metadata in SageMaker model format
+        return metaDataMap.containsKey(FRAMEWORK) &&
+                metaDataMap.containsKey(FRAMEWORK_VERSION) &&
+                metaDataMap.containsKey(MODEL) &&
+                metaDataMap.containsKey(MODEL_TYPE) &&
+                metaDataMap.containsKey(VERSION) &&
+                metaDataMap.containsKey(TASK);
+    }
+
+    private static Boolean hasCustomizedSchema(Map<String, Object> metaDataMap) throws ModelException {
+        if (metaDataMap.containsKey(SCHEMA) && !metaDataMap.get(SCHEMA).toString().contains(YAML_FORMAT)) {
+            if (!metaDataMap.containsKey(SCHEMA_HMAC)) {
+                logger.error("Invalid SageMaker Model format due to missing SchemaHmac when customized Schema is defined");
+                throw new ModelException("Invalid SageMaker Model format");
+            }
+            // todo: validate schema hmac
+            return true;
+        }
+        return false;
+    }
+
+    private static Boolean hasCustomizedInferenceSpec(Map<String, Object> metaDataMap) throws ModelException {
+        if (metaDataMap.containsKey(INFERENCE_SPEC)) {
+            if (!metaDataMap.containsKey(INFERENCE_SPEC_HMAC)) {
+                logger.error("Invalid SageMaker Model format due to missing InferenceSpecHmac when customized InferenceSpec is defined");
```

Comment:
error logging is unnecessary, we catch exception at higher level and log the message